src/inspector/index.ts

```diff
@@ -0,0 +1,66 @@
+import { Context, Integrations, LegacySettings } from '..'
+
+// FIXME: Import these types directly from `@segment/inspector-core`
+type SegmentEvent = object
+type Inspector = {
+  start: (config: object) => void
+  trace: (eventPhase: object) => void
+}
+
+export const inspectorHost = (() => {
```

Comment:
I'm curious why you used an IIFE to set the value of `inspectorHost`. Since this file is already a module you could just export `connectInspector` directly.

Thinking about this more, it might be useful to have `inspectorHost` be a function that takes an `Analytics` object. Then you can rely on analytics to resolve some of your values for you, for example: `analytics.user.id()` instead of `localStorage.getItem('ajs_user_id')`. The benefit for this example would be you'd get the correct user id from analytics itself, be it stored in cookies, localstorage, or in memory!

Reply:
Those are actually pretty good points!

I'll experiment with binding the host directly to an Analytics instance when I get back from vacation.

_(You too are welcome to try it out! there are tests already which will ensure we don't mess up compliance)_